### PR TITLE
Add prometheus (metrics) in bragi

### PR DESCRIPTION
**Pull request by AzHicham**

### CODE REVIEW

File: libs/mimir/src/adapters/primary/bragi/api.rs [hunk 1]
```diff
@@ -230,6 +230,17 @@ macro_rules! status {
 }
 pub use status;
 
+#[macro_export]
+macro_rules! metrics {
+    ($cl:expr, $pt:expr) => {
+        routes::metrics()
+            .and(routes::with_client($cl))
```
Comment (crocme10) [inline libs/mimir/src/adapters/primary/bragi/api.rs:237]: You don't need the client here

File: libs/mimir/src/adapters/primary/bragi/handlers.rs [hunk 2]
```diff
@@ -250,6 +251,21 @@ where
     }
 }
 
+pub async fn metrics<S>(
+    _client: S,
```
Comment (crocme10) [inline libs/mimir/src/adapters/primary/bragi/handlers.rs:255]: so you can remove the client here

File: src/bragi/server.rs [hunk 3]
```diff
@@ -105,17 +107,21 @@ pub async fn run_server(settings: Settings) -> Result<(), Error> {
...
+    let prometheus = PrometheusMetrics::default();
+
     // Here I place reverse_geocoder first because its most likely to get hit.
     let api = reverse_geocoder!(client.clone(), settings.query.clone())
         .or(forward_geocoder!(client.clone(), settings.query.clone()))
         .or(features!(client.clone(), settings.query.clone()))
         .or(forward_geocoder_explain!(client.clone(), settings.query))
-        .or(status!(client, &settings.elasticsearch.url))
+        .or(status!(client.clone(), &settings.elasticsearch.url))
+        .or(metrics!(client, prometheus.clone()))
```
Comment (crocme10) [inline src/bragi/server.rs:118]: no client here

File: libs/mimir/Cargo.toml [hunk 4]
```diff
@@ -48,6 +48,7 @@ tracing-log = "0.1.2"
 tracing-subscriber = "0.2.17"
 url = { version = "2.2", features = [ "serde" ] }
 warp = "0.3.2"
+prometheus = "0.13.0"
```
Comment (crocme10) [inline libs/mimir/Cargo.toml:51]: Maybe prometheus should be a feature.... not everyone needs it

Comment (AzHicham) [reply libs/mimir/Cargo.toml]: done

**Review by azime — APPROVED** [review]

**Pull request merged**